pallets/liquid-staking/src/lib.rs

```diff
@@ -788,6 +771,27 @@ pub mod pallet {
             Ok(())
         }
 
+        #[require_transactional]
+        fn do_update_exchange_rate(bonding_amount: BalanceOf<T>) -> DispatchResult {
+            let old_exchange_rate = Self::exchange_rate();
+            let matching_pool = Self::matching_pool();
+            match Rate::checked_from_rational(
+                bonding_amount
+                    .checked_add(matching_pool.total_stake_amount)
+                    .ok_or(ArithmeticError::Overflow)?,
+                T::Assets::total_issuance(Self::liquid_currency()?)
+                    .checked_add(matching_pool.total_unstake_amount)
+                    .ok_or(ArithmeticError::Overflow)?,
```

Comment:
@MrPai  will it cause issues if we use the edited matching_ledger to update exchange_rate? if yes I'll still need to pass old matching_pool to this function

Reply:
We didn't think about what was matched

Reply:
sorry can you elaborate more?

Reply:
ok so you still want to use old matching_ledger right?

Reply:
`old_matching_ledger` looks clear for now, however  edited matching_ledger maybe confused.